libr/anal/cc.c

```diff
@@ -139,19 +136,18 @@ R_API char *r_anal_cc_to_string (RAnal *anal, RAnalCC* cc) {
 				 snprintf (buf, sizeof (buf),
 					"0x%"PFMT64x, cc->args[cc->nargs-i]);
 			else strncpy (buf, "unk", sizeof (buf)-1);
-			buf_len = strlen (buf);
-			if ((buf_len+str_len+5)>=sizeof (str)) {
-				strcat (str, "...");
-				break;
+			str = r_str_concat (str, buf);
+			if (i<cc->nargs-1){
+				str = r_str_concat (str, ", ");
 			}
-			strcat (str, buf);
-			str_len += buf_len;
-			if (i<cc->nargs-1) strcat (str, ", ");
 		}
-		strcat (str, ")");
+		str = r_str_concat (str, ")");
 		break;
 	}
-	return strdup (str);
+	if(str){
```

Comment:
str is already in the heap no need to re-strdup't

also, you are leaking here

and there are missing spaces around () and {